### ISSUE TITLE
[Question]: 如何设置多卡pipline并行推理？

### ISSUE THREAD

**Issue by zhaogf01** [PaddlePaddle/PaddleNLP#8539]
### 请提出你的问题

# coding=utf-8
import paddle
from paddle.distributed import fleet
from paddlenlp.transformers import AutoTokenizer, AutoModelForCausalLM,AutoModelForCausalLMPipe
from paddlenlp.transformers.llama import LlamaTokenizer
from paddle.distributed.fleet.meta_parallel.pipeline_parallel import PipelineParallel

world_size = paddle.distributed.get_world_size()
pp_degree = world_size
tp_degree = 1
if world_size > 2:
    pp_degree = 2
    assert world_size % pp_degree == 0
    tp_degree = world_size // pp_degree

strategy = fleet.DistributedStrategy()
strategy.hybrid_configs = {
    "dp_degree": 1,
    "mp_degree": tp_degree,
    "pp_degree": pp_degree,
    "sharding_degree": 1,
}
fleet.init(is_collective=True, strategy=strategy)
hcg = fleet.get_hybrid_communicate_group()

if pp_degree > 1:
    model_class = xxxForCausalLMPipe
else:
    model_class = xxxForCausalLM

model_name_or_path = "/workspace/xxx"
model = model_class.from_pretrained(
    model_name_or_path,
    tensor_parallel_degree=tp_degree,
    tensor_parallel_rank=hcg.get_model_parallel_rank(),
    tensor_parallel_output=False,
    dtype="float32"
    # use_flash_attention=True,
)

model.eval()

tokenizer = LlamaTokenizer.from_pretrained(model_name_or_path, add_eos_token=False, add_bos_token=False, eos_token='<eod>')
tokenizer.add_tokens(['<sep>', '<pad>', '<mask>', '<predict>', '<FIM_SUFFIX>', '<FIM_PREFIX>', '<FIM_MIDDLE>','<commit_before>','<commit_msg>','<commit_after>','<jupyter_start>','<jupyter_text>','<jupyter_code>','<jupyter_output>','<empty_output>'], special_tokens=True)

if pp_degree > 1:
    pp_model = PipelineParallel(layers=model, hcg=hcg, strategy=strategy)

    input_features = tokenizer("长江多长？", return_tensors="pd")
    print(input_features)
    ret = pp_model.eval_batch(data=[input_features["input_ids"], input_features["position_ids"]], compute_loss=False)
    print(tokenizer.batch_decode(ret.item()))

我采用了这种方式，有个报错
![image](https://github.com/PaddlePaddle/PaddleNLP/assets/148510471/1cd23b4b-c7d4-4c7a-8d48-a8085dcdcfb7)
请问如何开启流水线并行

**Comment by ZHUI** [comment]
一般使用 TP 推理，流水线推理的话暂时不支持
